src/apm/health.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package apm
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/newrelic/k8s-agents-operator/src/api/v1alpha2"
+)
+
+const (
+	envHealthFleetControlFilepath = "NEW_RELIC_FLEET_CONTROL_HEALTH_PATH"
+	envHealthListenPort           = "NEW_RELIC_SIDECAR_LISTEN_PORT"
+	healthSidecarContainerName    = "newrelic-apm-health"
+	healthVolumeName              = "newrelic-apm-health"
+
+	HealthInstrumentedAnnotation = "newrelic.com/apm-health"
+)
+
+var (
+	defaultHealthListenPort = 6194
+)
+
+var healthDefaultEnv = []corev1.EnvVar{
+	{Name: envHealthListenPort, Value: fmt.Sprintf("%d", defaultHealthListenPort)},
+}
+
+func (i *baseInjector) injectHealth(ctx context.Context, inst v1alpha2.Instrumentation, ns corev1.Namespace, pod corev1.Pod) (corev1.Pod, error) {
```

Comment:
I think we should do a pointer to corev1.Pod in and pointer to the corev1.Pod out.
`func (i *baseInjector) injectHealth(ctx context.Context, inst v1alpha2.Instrumentation, ns corev1.Namespace, pod *corev1.Pod) (*corev1.Pod, error)`